lib/src/files.rs

```diff
@@ -404,8 +372,11 @@ mod tests {
         );
         // One side unchanged, two other sides make the same change
         assert_eq!(
-            merge(&[b"a", b"a"], &[b"", b"a", b""]),
-            MergeResult::Resolved(b"".to_vec())
+            merge(&[b"a", b"a"], &[b"b", b"a", b"b"]),
+            MergeResult::Conflict(vec![MergeHunk::Conflict(ConflictHunk {
+                removes: vec![b"a".to_vec(), b"a".to_vec()],
+                adds: vec![b"b".to_vec(), b"a".to_vec(), b"b".to_vec()]
+            })])
         );
         // One side unchanged, two other sides make the different change
         assert_eq!(
```

Comment:
Unless I missed it, let's add a test for 

    merge(&[b"c", b"d"], &[b"a", b"a", b"a"])

I wonder if it'll change similarly to the test above. I'm not sure whether changing these tests is good (haven't made up my mind yet).

This is related to https://github.com/martinvonz/jj/pull/1610#discussion_r1199823932

Reply:
There's a test for that in `merge.rs` (I think I enumerated all possible cases with up to 2 removes), but I've added one here too.

Reply:
I was more interested in whether, before this commit, that test would work the same as after. I'll check it on my machine.

Reply:
I did check. It's the same before and after.

If you feel like it, you could create an extra commit to demonstrate that.

I was worried about the changed test, but on second thought the change in behavior should be rarely noticeable, since file content conflicts are best resolved by tackling them one 3-sided conflict at a time. This might be nice to mention in the commit description.